src/transformers/modeling_tf_t5.py

```diff
@@ -1422,6 +1425,10 @@ def prepare_inputs_for_generation(self, inputs, past, attention_mask, use_cache,
         else:
             encoder_outputs, past_key_values = past[0], past[1]
 
+        # cut decoder_input_ids if past is used
+        if past_key_values is not None:
+            inputs = inputs[:, -1:]
+
```

Comment:
Isn't that your bug number 3?

Reply:
True! I forgot to remove it => Will write a test for it now

Reply:
No this is actually correct => it's for generation. Writing tests helps anyways though